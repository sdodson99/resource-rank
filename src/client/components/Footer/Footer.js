import React from 'react';
import styles from './Footer.module.css';
import Image from 'next/image';

const Footer = () => (
  <footer className={styles.Footer} data-testid="Footer">
    <div className="content-container">
      <div className="sm:flex text-center sm:text-left">
        <div className="order-1 sm:order-2 sm:ml-10">
          <h1 className="text-5xl">Our Mission</h1>
          <p className="py-3">
            is to <strong>simplify</strong> your learning journey. Learning can
            be <strong>difficult</strong>, especially when starting. There are{' '}
            <strong>millions</strong> of educational courses, videos, and books{' '}
            <strong>scattered</strong> across the Internet.
          </p>
          <p className="py-3">
            Resource Rankâ€™s goal is to <strong>consolidate</strong> all of these
            resources so that you can <strong>quickly</strong> find the{' '}
            <strong>best</strong> resources for what you are learning.
          </p>
        </div>
        <div className="order-2 sm:order-1 mt-10 sm:mt-0">
          <Image
            src="/img/logo.svg"
            alt="Resource Rank Logo"
            height="75"
            width="75"
            layout="fixed"
            priority={true}
          />
        </div>
      </div>
    </div>
  </footer>
);

Footer.propTypes = {};

Footer.defaultProps = {};

export default Footer;
