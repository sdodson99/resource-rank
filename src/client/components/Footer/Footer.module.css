.Footer {
  background-color: var(--color-primary);

  @apply py-10 text-white text-lg;
}
