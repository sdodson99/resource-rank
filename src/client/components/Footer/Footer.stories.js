import React from 'react';
import Footer from './Footer';

export default {
  component: Footer,
  title: 'Components/Footer',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <Footer {...args} />;

export const Default = Template.bind({});
Default.args = {};
