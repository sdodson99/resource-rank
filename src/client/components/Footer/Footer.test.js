import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Footer from './Footer';
import renderer from 'react-test-renderer';

describe('<Footer />', () => {
  it('should mount', () => {
    render(<Footer />);

    const footer = screen.getByTestId('Footer');

    expect(footer).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const tree = renderer.create(<Footer />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
