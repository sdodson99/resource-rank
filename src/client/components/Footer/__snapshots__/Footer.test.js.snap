// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
<footer
  className="Footer"
  data-testid="Footer"
>
  <div
    className="content-container"
  >
    <div
      className="sm:flex text-center sm:text-left"
    >
      <div
        className="order-1 sm:order-2 sm:ml-10"
      >
        <h1
          className="text-5xl"
        >
          Our Mission
        </h1>
        <p
          className="py-3"
        >
          is to 
          <strong>
            simplify
          </strong>
           your learning journey. Learning can be 
          <strong>
            difficult
          </strong>
          , especially when starting. There are
           
          <strong>
            millions
          </strong>
           of educational courses, videos, and books
           
          <strong>
            scattered
          </strong>
           across the Internet.
        </p>
        <p
          className="py-3"
        >
          Resource Rankâ€™s goal is to 
          <strong>
            consolidate
          </strong>
           all of these resources so that you can 
          <strong>
            quickly
          </strong>
           find the
           
          <strong>
            best
          </strong>
           resources for what you are learning.
        </p>
      </div>
      <div
        className="order-2 sm:order-1 mt-10 sm:mt-0"
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": 75,
              "overflow": "hidden",
              "position": "relative",
              "width": 75,
            }
          }
        >
          <img
            alt="Resource Rank Logo"
            decoding="async"
            src="/_next/image?url=%2Fimg%2Flogo.svg&w=256&q=75"
            srcSet="/_next/image?url=%2Fimg%2Flogo.svg&w=96&q=75 1x, /_next/image?url=%2Fimg%2Flogo.svg&w=256&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</footer>
`;
